plugins/DevicesDetection/Columns/DeviceModel.php

```diff
@@ -33,7 +34,23 @@ public function onNewVisit(Request $request, Visitor $visitor, $action)
         $userAgent = $request->getUserAgent();
         $parser    = $this->getUAParser($userAgent);
 
-        return $parser->getModel();
+        $model = $parser->getModel();
+
+        if (!empty($model)) {
+            return $model;
+        }
+
+        $deviceType = $parser->getDeviceName();
+
+        if (!empty($deviceType)) {
+            return 'generic ' . $deviceType;
+        }
+
+        if ($parser->isMobile()) {
+            return 'generic mobile';
+        }
+
+        return '';
```

Comment:
This will also only affect new data, right? Would it be possible to do this during archiving so it can be applied to old reports?

Don't think this is super important if not, though.

Reply:
It's currently only for new data. Doing that while archiving would be possible but maybe a bit slow, as we would need to run a custom archiving query having an `CASE` statement to set the generic name based on the device type field... Also it might be a bit weird to have reports showing a `generic` model, but none can be found with segmentation as the log tables were not updated.
So maybe the correct way would be to write some kind of reattribution script like for geolocation.